cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/BulkWriterConfig.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cassandra.spark.bulkwriter;
+
+import java.io.Serializable;
+
+import org.apache.spark.sql.types.StructType;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Immutable configuration data class for BulkWriter jobs that is safe to broadcast to Spark executors.
+ * This class contains pre-computed, serializable values that were computed on the driver.
+ * <p>
+ * Serialization Architecture:
+ * This class is the ONLY object that gets broadcast to Spark executors (via Spark's broadcast mechanism).
+ * It contains serializable implementations of cluster information ({@link SerializableClusterInfo} or
+ * {@link SerializableClusterInfoGroup}) that have zero transient fields for safe serialization.
+ * <p>
+ * On the driver, {@link BulkWriterContext} instances use driver-only implementations like
+ * {@link CassandraClusterInfo}. Before broadcasting, these are converted to serializable forms.
+ * On executors, {@link BulkWriterContext} instances are reconstructed from this config using
+ * {@link BulkWriterContext#from(BulkWriterConfig, boolean)}, which uses the serializable cluster
+ * information directly without converting back to driver-only types.
+ */
+public final class BulkWriterConfig implements Serializable
```

Comment:
My concern here is that some of these fields need to be recomputed dynamically from the executor and we are not honoring the when we are trying to fetch non-cached values in some of the methods of this class.

Reply:
addressed